flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/plan/stream/sql/agg/OverAggregateTest.scala

```diff
@@ -480,4 +482,31 @@ class OverAggregateTest extends TableTestBase {
     util.verifyExecPlan(sql)
   }
 
+  @Test
+  def testWindowBoundaryNotNumeric(): Unit = {
+    val sql =
+      """
+        |SELECT c,
+        |    COUNT(a) OVER (PARTITION BY b ORDER BY proctime
+        |        ROWS BETWEEN '2' PRECEDING AND CURRENT ROW) AS cnt1
+        |FROM MyTable
+      """.stripMargin
+
+    assertThatThrownBy(() => util.verifyExecPlan(sql))
+      .hasRootCauseInstanceOf(classOf[ValidationException])
```

Comment:
Please also match against a root cause. Not just the pure type which could be anything.